pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -136,24 +135,19 @@ def get_receipt(self, payment_account: PaymentAccount, pay_response_url: str, in
         # If pay_response_url is present do all the pre-check, else check the status by using the invoice id
         if pay_response_url is not None:
             parsed_args = parse_url_params(pay_response_url)
-            if not parsed_args or parsed_args.get('hashValue', None) is None:
-                raise BusinessException(Error.DIRECT_PAY_INVALID_RESPONSE)
 
             # validate if hashValue matches with rest of the values hashed
-            hash_value = parsed_args.pop('hashValue', None)[0]
+            hash_value = parsed_args.pop('hashValue', None)
             pay_response_url_without_hash = urlencode(parsed_args)
-            if not HashingService.is_valid_checksum(pay_response_url_without_hash, hash_value):
-                current_app.logger.warn(f'Hash not matching for pay response URL : {pay_response_url}')
-                raise BusinessException(Error.DIRECT_PAY_INVALID_RESPONSE)
 
             # Check if trnApproved is 1=Success, 0=Declined
-            trn_approved: str = parsed_args.get('trnApproved')[0]
-            if trn_approved != '1':
-                current_app.logger.info('Returning the receipt as the transaction is not successful')
+            trn_approved: str = parsed_args.get('trnApproved')
+            if trn_approved == '1' and not HashingService.is_valid_checksum(pay_response_url_without_hash, hash_value):
```

Comment:
should this If be separate? the logging might confuse if the trn_approved something else and log will print not valid checksum

Reply:
Ok, will do